src/Context/HelperContext.php

```diff
@@ -148,29 +160,38 @@ public function takeScreenshotAfterFailedStep(AfterStepScope $scope) {
       $featureFolder = str_replace(' ', '', $scope->getFeature()->getTitle());
 
       // Get the name of the failed test.
-      $scenarioName = $this->currentScenario->getTitle();
-      $failedTest = str_replace(' ', '', $scenarioName);
+      $failedTest = str_replace(' ', '', $this->scenarioName);
 
-      // Set the screenshot folder.
+      // Set the destination screenshot folder.
       $filePath = $this->parameters['screenshot_path'] . '/' . $featureFolder;
       if (!file_exists($filePath)) {
         mkdir($filePath);
       }
 
+      //  Headless test - save an HTML dump.
       if ($driver instanceof \Behat\Mink\Driver\BrowserKitDriver) {
         $html_data = $this->getSession()->getDriver()->getContent();
-        $fileName = date(self::DATE_FORMAT_CONCISE) . '-' . $failedTest . '.html';
+        $fileName = date(self::DATE_FORMAT_CONCISE) . '-' . $failedTest . self::EXTENSION_HTML;
         file_put_contents($filePath . '/' . $fileName, $html_data);
         return;
       }
 
+      //  Browser test - save an HTML dump + a JPG.
       if ($driver instanceof \Behat\Mink\Driver\Selenium2Driver) {
-        $fileName = date(self::DATE_FORMAT_CONCISE) . '-' . $failedTest . '.jpg';
+
+        //  Generate HTML dump.
+        $html_data = $this->getSession()->getDriver()->getContent();
+        $fileName = date(self::DATE_FORMAT_CONCISE) . '-' . $failedTest . self::EXTENSION_HTML;
```

Comment:
Don't repeat code if possible. The date is reused 3 times, put it in a var and reuse.